cmd/ipfs-cluster-service/main.go

```diff
@@ -580,15 +579,69 @@ func run(c *cli.Context) error {
 	return nil
 }
 
-func setupLogLevel(lvl string) {
-	for f := range ipfscluster.LoggingFacilities {
-		ipfscluster.SetFacilityLogLevel(f, lvl)
+func setupLogLevel(debug bool, compLogLevel []string) error {
+	// if debug is set to true, log everything in debug level
+	if debug {
+		ipfscluster.SetFacilityLogLevel("*", "DEBUG")
+		return nil
+	}
+
+	var logLevel string
```

Comment:
This is overcomplicated from this point.

0. Make the compLogFacs maps and set "all" -> defaultLogLevel
1. Loop and parse the logLevel flags into additional entries in the map or overwrite "all" when not component-specific.
2. for f := range ipfscluster.LoggingFacilities ... -> set to compLogFacs["all"]
3. Loop compLogFacs and set the rest of things.

The fact that 2 of your case entries are nearly identical should raise the flag that a refactor is needed.